core/prep.js

```diff
@@ -109,8 +109,9 @@ class Prep {
     doc.docType = 'file'
     assignId(doc)
     assignId(was)
-    let docIgnored = this.ignore.isIgnored(doc)
-    let wasIgnored = this.ignore.isIgnored(was)
+    let docIgnored = this.ignore.isIgnored(doc._id, 'file')
+    const wasKind = was.docType === 'folder' ? 'directory' : was.docType
+    let wasIgnored = this.ignore.isIgnored(was._id, wasKind)
```

Comment:
The old code looks more simple than the new one...

Reply:
I know, this is the problem with transitions :/. I didn't want to go too far but if that's an issue I can change my position on this.